upstairs/src/downstairs.rs

```diff
@@ -3166,6 +3174,30 @@ impl Downstairs {
             self.ackable_work.insert(ds_id);
         }
 
+        if job.reply_time.iter().all(Option::is_some) && !job.replay {
+            let fastest_id = ClientId::iter()
+                .min_by_key(|i| job.reply_time[*i].unwrap())
+                .unwrap();
+            let slowest_time = *job.reply_time.iter().flatten().max().unwrap();
+
+            // Apply a delay to the fastest client, and clear the delay for the
+            // other two clients.
+            for i in ClientId::iter() {
+                let delay_time_us: u64 = if i == fastest_id {
+                    // 0 delay below 10ms of lead time, then linearly increasing
+                    // to a maximum of 10 ms.  These are all roughly-eyeballed
+                    // numbers!
+                    let dt = slowest_time - job.reply_time[i].unwrap();
+                    let lead_time_us: u64 = dt.as_micros().try_into().unwrap();
+                    (lead_time_us.saturating_sub(10_000) / 100).min(10_000)
+                } else {
+                    0
+                };
+
+                self.clients[i].set_delay_us(delay_time_us);
+            }
+        }
```

Comment:
If we're not setting the per-client time, should we clear it? What if one Downstairs is `Faulted`, or in replay - the previously set delays will still apply.

Reply:
> If we're not setting the per-client time, should we clear it?

I don't think so.  If we did that, then the first two replies to a job would clear the delay, then the third reply would set it (before it's cleared again by the first reply to the next job).

> What if one Downstairs is Faulted, or in replay - the previously set delays will still apply.

I added logic to clear the per-client delay in `Client::reinitialize` (1190fa30237afd358ff98f397d78ea82e8297014).

Maybe we should clear it if _any_ of the clients restart?  Otherwise, I could imagine a situation where 

- DS1 is temporarily slower than DS2, so we delay DS2
- Then DS3 returns an error, so we restart it
- We're still delaying DS2, and that won't change until DS3 comes back online

Reply:
> I don't think so. If we did that, then the first two replies to a job would clear the delay, then the third reply would set it (before it's cleared again by the first reply to the next job).

:+1:

> Maybe we should clear it if any of the clients restart?

That sounds like a good idea, yeah.

Reply:
Okay, done in 80b6f734175c2bc8e222c8d6f7875e53cbd5c9ca